### ISSUE TITLE
SMTPClient documentation doesn't describe sending multiple emails

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10444]
|[<img alt="bitglue's avatar" src="https://avatars.githubusercontent.com/u/1232472?s=50" width="50" height="50">](https://github.com/bitglue)| @bitglue reported|
|-|-|
|Trac ID|trac#684|
|Type|defect|
|Created|2004-08-21 06:13:38Z|

Attachments:

* [smtpclient.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a79/a79e9a409bded1928e5dad9765d53e7bce91d555/9e9778ee45daea3a30b259340518a7444f14cf9e.diff) (1187 bytes) - added by drake on 2009-05-02 19:50:14Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__684 684
type__defect defect
reporter__indigo indigo
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__documentation documentation
time__1093068818000000 1093068818000000
changetime__1241716850000000 1241716850000000
version__ 
owner__ 
cc__Jerub cc__hypatia cc__indigo cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="bitglue's avatar" src="https://avatars.githubusercontent.com/u/1232472?s=50" width="50" height="50">](https://github.com/bitglue)|@bitglue commented|
|-|-|

```
#!html
<pre>
http://twistedmatrix.com/documents/current/api/twisted.protocols.smtp.SMTPClient.html
Describes getMailFrom, getMailTo, and getMailData as rather simply returning.
However, if getMailFrom does not return None, SMTPClient sends the mail and
returns to the DATA state, calling getMailFrom, then getMailTo then getMailData
again. The ability to send multiple mails per connection is not obvious, and if
one were to follow the documentation, the result would be an endless loop. Now I
know who is responsible for all those duplicate posts on mailing lists!
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Jerub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Jerub commented|
|-|-|

```
#!html
<pre>
wow, this is over a year old. I'm stealing this issue.

The link is a 404 now, the new link is:
http://twistedmatrix.com/documents/current/api/twisted.mail.smtp.SMTPClient.html
because of the package split.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="drake's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|drake removed owner<br>drake set status to `new`|
|-|-|

The attached patch adds the requisite detail to the doc string.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @exarkun|
|-|-|

Looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [26894]) Apply smtpclient.diff adding docs for SMTPClient and SMTPSender

Author: drake, itamar
Reviewer: exarkun
Fixes: [#10444](https://github.com/twisted/twisted/issues/10444)

Add documentation to `SMTPClient` about how `getMailFrom`, `getMailTo`,
and `getMailData` are called, possibly many times, allowing multiple
emails to be sent.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
